### PR TITLE
add unbookmark handler

### PR DESCRIPTION
**Pull request by jmilldotdev**
## What does this PR do and why?

_Provide a description of what this PR does. Link to any relevant GitHub issues, Notion tasks or Discord discussions._

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by jmilldotdev** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2280** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2280" target="_blank">Graphite</a> 
    * **PR #2281** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2281" target="_blank">Graphite</a> 
      * **PR #2282** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2282" target="_blank">Graphite</a> 
        * **PR #2283** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2283" target="_blank">Graphite</a> 
          * **PR #2284** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2284" target="_blank">Graphite</a> 
            * **PR #2285** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2285" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2285?utm_source=stack-comment).

File: src/components/Project/ProjectHeader/BookmarkProjectButton.tsx [hunk 1]
```diff
@@ -34,11 +35,28 @@ const BookmarkProjectButton = () => {
       await axios.post(`/api/bookmark`, data)
     } catch (error) {
       console.error(error)
+      emitErrorNotification(t`Failed to bookmark project`)
```
Comment (tomquirk) [inline src/components/Project/ProjectHeader/BookmarkProjectButton.tsx:38]: aylmao

File: src/pages/api/bookmark/index.page.ts [hunk 2]
```diff
@@ -1,27 +1,39 @@
...
   projectId: number
   cv: CV
+  shouldUnbookmark?: boolean
 }
 
 interface BookmarkProjectRequest extends NextApiRequest {
   body: BookmarkProjectData
 }
 
 const handler = async (req: BookmarkProjectRequest, res: NextApiResponse) => {
```
Comment (tomquirk) [inline src/pages/api/bookmark/index.page.ts:19]: **issue** one thing with this... as it is currently, can anyone bookmark a project on behalf of anyone else?

I'm assuming that's an issue but might be missing something. lmk

If it is, i guess we'll need some form of API auth... 😓

**Review by tomquirk — CHANGES_REQUESTED** [review]
one question here

File: src/pages/api/bookmark/index.page.ts (hunk 2 above)
Comment (jmilldotdev) [reply src/pages/api/bookmark/index.page.ts:19]: yes, this is true. the idea is to smoke test it for now, then add a session to browsing the website via signing a message in a future PR. this is what i plan to do next.

**Pull request merged**